### PR TITLE
Added explanations for autofocus and onSubmit to TextField/TextArea

### PR DESCRIPTION
**Pull request by btraut**
I had questions why `TextArea` and `TextField` didn't have `onSubmit` or `autofocus` in the past. Rather than keeping the answers to those questions myself, I figured I'd add it to the docs.

### CODE REVIEW

File: src/TextArea/TextArea.doc.js [hunk 1]
```diff
@@ -170,3 +170,26 @@ card(
...
+    autofocus interrupts normal page flow for screen readers making it an
+    anti-pattern for accessibility.
+  `
+);
+
+card(
+  'onSubmit',
+  md`
+    \`TextArea\` is commonly used as an input in forms along side submit buttons.
+    In these cases, users will expect the pressing Enter or Return with the input
```
Comment (pfarejowicz) [inline src/TextArea/TextArea.doc.js]: `In these cases, users will expect the pressing Enter or Return` typo here, did you mean to use `that`?

File: src/TextField/TextField.doc.js [hunk 2]
```diff
@@ -197,3 +197,26 @@ card(
...
+    autofocus interrupts normal page flow for screen readers making it an
+    anti-pattern for accessibility.
+  `
+);
+
+card(
+  'onSubmit',
+  md`
+    \`TextField\` is commonly used as an input in forms along side submit buttons.
+    In these cases, users will expect the pressing Enter or Return with the input
```
Comment (pfarejowicz) [inline src/TextField/TextField.doc.js]: Same typo here I think

**Comment by chrislloyd** [comment]
🙏  bias towards action - awesome :)

File: src/TextArea/TextArea.doc.js (hunk 1 above)
Comment (btraut) [reply src/TextArea/TextArea.doc.js]: Great catch.

PSA: Don't try to grammar at 1am.

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please add a [changelog.md](https://github.com/pinterest/gestalt/blob/master/CHANGELOG.md) entry for your changes.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**